crates/cadenza-web/app/src/lib/monaco-cadenza.ts

```diff
@@ -0,0 +1,219 @@
+// Monaco Editor language configuration and LSP integration for Cadenza
+
+import * as monaco from 'monaco-editor';
+import type { CadenzaWasm, LspDiagnostic } from '../types/cadenza';
+
+// Language configuration for Cadenza
+export const CADENZA_LANGUAGE_ID = 'cadenza';
+
+export function registerCadenzaLanguage() {
+  // Register the language
+  monaco.languages.register({ id: CADENZA_LANGUAGE_ID });
+
+  // Set up language configuration
+  monaco.languages.setLanguageConfiguration(CADENZA_LANGUAGE_ID, {
+    comments: {
+      lineComment: '//',
+      blockComment: ['/*', '*/'],
+    },
+    brackets: [
+      ['{', '}'],
+      ['[', ']'],
+      ['(', ')'],
+    ],
+    autoClosingPairs: [
+      { open: '{', close: '}' },
+      { open: '[', close: ']' },
+      { open: '(', close: ')' },
+      { open: '"', close: '"' },
+      { open: "'", close: "'" },
+    ],
+    surroundingPairs: [
+      { open: '{', close: '}' },
+      { open: '[', close: ']' },
+      { open: '(', close: ')' },
+      { open: '"', close: '"' },
+      { open: "'", close: "'" },
+    ],
+  });
+
+  // Set up syntax highlighting
+  monaco.languages.setMonarchTokensProvider(CADENZA_LANGUAGE_ID, {
+    keywords: [
+      'let', 'fn', 'if', 'else', 'match', 'for', 'while', 'return',
+      'true', 'false', 'nil',
+    ],
+    operators: [
```

Comment:
Ideally this list is generated by the rust code so any time we add one it would show up here